crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -810,54 +810,84 @@ impl<'src> Parser<'src> {
         }
     }
 
-    /// See: <https://docs.python.org/3/reference/simple_stmts.html#grammar-token-python-grammar-assignment_stmt>
+    /// Parse an assignment statement.
+    ///
+    /// # Panics
+    ///
+    /// If the parser isn't positioned at an `=` token.
```

Comment:
I'm not sure if that's true because `parse_list` will call `is_at_list_element` which returns `false` if the parser isn't positioned at a `=` in which case it goes into error recovery.

Reply:
Hmm, yes but the method is only called when the parser sees the `=` token.